### PR TITLE
MNTOR-3378 - Data broker manual removal  

### PR DESCRIPTION
**Pull request by codemist**
# References:

Jira: MNTOR-3378
Figma: https://www.figma.com/design/DTbmXzCQCw2PxXpHQW8yfW/Monitor-MVP-Enhancements?node-id=3809-6501&t=EniBV7WqvPVnuivk-4

Path: /user/dashboard/fix/data-broker-profiles/removal-under-maintenance

# Description

# Screenshot (if applicable)
image
image
image

Not applicable.

# How to test

# Checklist (Definition of Done)

- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: locales/en/exposure-card.ftl
Comment (flodolo) [inline locales/en/exposure-card.ftl:35]: Why marking this as deprecated, given the code is removing the usage? The string should be removed.

File: locales-pending/dashboard-premium.ftl
Comment (flodolo) [inline locales-pending/dashboard-premium.ftl:220]:
```suggestion
data-broker-removal-maintenance-estimated-time =
    { $range ->
        [one] Est. time to complete: { $range } minute
       *[other] Est. time to complete: { $range } minutes
    }
```

File: locales-pending/dashboard-premium.ftl
Comment (flozia) [inline locales-pending/dashboard-premium.ftl:214]: Nit: The designs are not showing a `.` here.

```suggestion
data-broker-removal-maintenance-header = { -brand-monitor } found this profile on <link_to_data_broker>{ $data_broker_name }</link_to_data_broker>
```

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/View.tsx
Comment (flozia) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/View.tsx:237]: After resolving an exposure we need to show [this indicator](https://www.figma.com/design/DTbmXzCQCw2PxXpHQW8yfW/Monitor-MVP-Enhancements?node-id=3809-6687&node-type=instance&m=dev) on the respective exposure card.

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx
Comment (flozia) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx:103]: This header would need to be:

```suggestion
              "data-broker-removal-maintenance-rationale-header",
```

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx
Comment (flozia) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx:216]: The only way to go back to the exposures when the “removal instructions” are open is by clicking “Back to exposures” (which for many viewport sizes is out of sight). I’m wondering if it would not be more intuitive to follow our existing pattern of showing explainers in dialogs. If that was already decided against by UX and/or is out of scope we should at least make sure that the closing “x” button does not link back to the dashboard.

image

File: src/app/components/client/exposure_card/ExposureCard.test.tsx
Comment (flozia) [inline src/app/components/client/exposure_card/ExposureCard.test.tsx:90]: Nit: Should we also add a test that checks that the description is _not_ shown for other statuses?

**Review by flozia — COMMENTED** [review]
This is looking good so far! There are some open comments and questions I’m having:

- Before landing any parts of this feature we’ll need to make sure to guard it behind a feature flag.
- Additionally, we would need to add telemetry (potentially in a follow-up) to the CTAs to fulfill the acceptance criteria.
- For smaller viewports it might be better to stack the buttons:
image

**Review by flodolo — APPROVED** [review]

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/page.tsx
Comment (flozia) [reply src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/page.tsx]: @codemist From what I understand it works like this:
- Passing the second optional argument `stepId` makes `getNextGuidedStep` “look” for the next step from where we currently are in the guided flow.
- If we don’t pass the current step, the function checks for the first relevant step for the user.

So I think this is the expected behaviour?

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx
Comment (flozia) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx:63]: It is safer to handle potential errors here when calling `fetch`. If this function throws an error we can’t recover. Let’s wrap the body of `handleManualRemovalChange` in a `try...catch`.

File: src/app/components/client/exposure_card/ExposureCardDataClass.tsx
Comment (flozia) [inline src/app/components/client/exposure_card/ExposureCardDataClass.tsx:125]: We can also remove the style for `.exposureTypeIcon` [here](https://github.com/mozilla/blurts-server/blob/mntor-3378/src/app/components/client/exposure_card/ExposureCardDataClass.module.scss#L10).

File: src/app/components/client/exposure_card/ScanResultCard.tsx
Comment (flozia) [reply src/app/components/client/exposure_card/ScanResultCard.tsx:127]: Looking at the exposure cards on the dashboard I’m currently only seeing a total count and not a list of items. Is this expected?

image

**Review by flozia — COMMENTED** [review]
Thanks for addressing my previous comments and making all the updates after aligning with UX!

I’m already submitting my “in progress” review to not block you waiting on reviews. At the moment I’m not able to access all views and test the functionality fully. I’m having issues to access the page `/user/dashboard/fix/data-broker-profiles/removal-under-maintenance`. I’ve tried adding mock broker results and also modifying the query in `getScanResultsWithBrokerUnderMaintenance` to get results back. Could you update the description with some guidance on how to test this best?

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx
Comment (flozia) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx:60]: The timeout here will always fire. Regardless of the status of the fetch operation on `L65`. Let’s check and see if we can find a solution for this.

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx
Comment (flozia) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx:107]: This `useEffect` should not be needed. We can directly set the `firstScanResultNotResolved` from within `handleManualRemovalChange` instead of storing the next scan result intermediately in its own state.

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx
Comment (flozia) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx:96]: Calling this function here will only set the loading state after the actual fetch request to resolve the result is done. In that time period a user could still click the button below again and trigger this function. We can do two thing to prevent this:
- check if `isLoadingNextDataBroker` is true at the beginning of `handleManualRemovalChange` and return early if that is the case
- and set call `setIsLoadingNextDataBroker(true)` directly after

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx
Comment (flozia) [reply src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx:60]: As discussed during the pairing session PR #5341 has a proposal for coordinating the fetch and timeout together.

**Review by flozia — APPROVED** [review]
Code looks good and works as expected.

**Pull request merged**
